core/modules/node/node.module

```diff
@@ -1641,9 +1641,8 @@ function node_menu() {
     'type' => MENU_VISIBLE_IN_BREADCRUMB,
     'file' => 'node.types.inc',
   );
-
   $items['node'] = array(
-    'page callback' => 'node_page_default',
+    'page callback' => _node_path_or_not(),
```

Comment:
Rather than two callbacks, could we just put the logic for the redirect directly into `node_page_default()`?